src/sources/util/http.rs

```diff
@@ -0,0 +1,113 @@
+use crate::event::Event;
+use bytes::Buf;
+use futures::{sync::mpsc, Future, Sink};
+use serde::Serialize;
+use std::error::Error;
+use std::fmt::{self, Display};
+use std::net::SocketAddr;
+use std::sync::{Arc, Mutex};
+use stream_cancel::Tripwire;
+use warp::filters::BoxedFilter;
+use warp::http::{HeaderMap, StatusCode};
+use warp::{Filter, Rejection};
+
+#[derive(Serialize, Debug)]
+pub struct ErrorMessage {
+    code: u16,
+    message: String,
+}
+impl ErrorMessage {
+    pub fn new(code: StatusCode, message: String) -> Self {
+        ErrorMessage {
+            code: code.as_u16(),
+            message,
+        }
+    }
+}
+impl Error for ErrorMessage {}
+impl Display for ErrorMessage {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}: {}", self.code, self.message)
+    }
+}
+
+fn string_to_static_str(s: String) -> &'static str {
+    //necessary because warp 0.1.18 needs a &'static str for the path
+    Box::leak(s.into_boxed_str())
```

Comment:
I don't think this is safe, and it definitely looks like a memory leak. https://doc.rust-lang.org/std/boxed/struct.Box.html#method.leak